irssi 1.0.0 fails to compile on Solaris 11 because OpenSSL is not compiled with EC support

Hi,

irssi 1.0.0 (0.8.20 was fine) does not compile on Solaris 11(.3) anymore:

```
  CC       network-openssl.o
"network-openssl.c", line 258: warning: statement not reached
"network-openssl.c", line 304: warning: statement not reached
"network-openssl.c", line 649: undefined symbol: EC_KEY
"network-openssl.c", line 649: undefined symbol: ec_key
"network-openssl.c", line 649: syntax error before or at: =
"network-openssl.c", line 662: warning: implicit function declaration: EVP_PKEY_get1_EC_KEY
"network-openssl.c", line 663: warning: implicit function declaration: EC_GROUP_get_curve_name
"network-openssl.c", line 663: warning: implicit function declaration: EC_KEY_get0_group
"network-openssl.c", line 664: warning: implicit function declaration: EC_KEY_free
cc: acomp failed for network-openssl.c
gmake[3]: *** [network-openssl.o] Error 2
gmake[3]: Leaving directory `/tmp/irssi-1.0.0/src/core'
```

Is there any chance of making the EC stuff optional? Because of patent issues Oracle has decided to ship base OpenSSL 1.0.2 in Solaris 11.3 without EC support…

--- a/src/core/network-openssl.c
+++ b/src/core/network-openssl.c
@@ -646,7 +646,11 @@ static void set_server_temporary_key_info(TLS_REC *tls, SSL *ssl)
 #ifdef SSL_get_server_tmp_key
 	// Show ephemeral key information.
 	EVP_PKEY *ephemeral_key = NULL;
+
+	// OPENSSL_NO_EC is for solaris 11.3 (2016), github ticket #598
+#ifndef OPENSSL_NO_EC
 	EC_KEY *ec_key = NULL;
+#endif
 	char *ephemeral_key_algorithm = NULL;
 	char *cname = NULL;
 	int nid;
@@ -658,6 +662,7 @@ static void set_server_temporary_key_info(TLS_REC *tls, SSL *ssl)
 				tls_rec_set_ephemeral_key_size(tls, EVP_PKEY_bits(ephemeral_key));
 				break;
 
+#ifndef OPENSSL_NO_EC
 			case EVP_PKEY_EC:
 				ec_key = EVP_PKEY_get1_EC_KEY(ephemeral_key);
 				nid = EC_GROUP_get_curve_name(EC_KEY_get0_group(ec_key));
@@ -670,6 +675,7 @@ static void set_server_temporary_key_info(TLS_REC *tls, SSL *ssl)
 
 				g_free_and_null(ephemeral_key_algorithm);
 				break;
+#endif
 
 			default:
 				tls_rec_set_ephemeral_key_algorithm(tls, "Unknown");

